docs/topics/jvm/jvm-records.md

```diff
@@ -59,12 +59,52 @@ To declare a data class with the `@JvmRecord` annotation, it must meet the follo
 * The class cannot be local.
 * The primary constructor of the class must be as visible as the class itself.
 
-### Enabling JVM records
+### Enable JVM records
 
 JVM records require the `16` target version or higher of the generated JVM bytecode.
 
 To specify it explicitly, use the `jvmTarget` compiler option in [Gradle](gradle-compiler-options.md#attributes-specific-to-jvm) or [Maven](maven.md#attributes-specific-to-jvm).
 
+## Annotate records in Kotlin
+
+To make an [annotation](annotations.md) available on both Kotlin properties **and** Java record components, add the
+following meta-annotations to your annotation declaration:
+
+* For Kotlin: [`@Target`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.annotation/-target/index.html)
+* For Java record components: [`@java.lang.annotation.Target`](https://docs.oracle.com/javase/8/docs/api/java/lang/annotation/Target.html)
+
+For example:
+
+```kotlin
+@Target(AnnotationTarget.CLASS, AnnotationTarget.PROPERTY)
+@java.lang.annotation.Target(ElementType.CLASS, ElementType.RECORD_COMPONENT)
+annotation class exampleClass
+```
+
+The `exampleClass` annotation can now be applied to Kotlin classes and properties, as well as Java classes and record
+components.
+
+### Propagate annotations of a record component
+
+<primary-label ref="experimental-general"/>
+
+<!-- Add warning about `@all` needing opt-in -->
+
+In Java, annotations on a record component are automatically propagated to the backing field, getter, and constructor parameter.
+You can replicate this behavior in Kotlin by using the [`@all:`](annotations.md) use-site target.
```

Comment:
This may be purely a presentation issue, but as I understand it, the use-site target is `all`, not `@all` since `@` is a part of the annotation. At least on the linked page on annotations we mention them without the `@`.

The current version is more evocative I'd say, but it may obscure the syntax.